Index: web/webapp/src/main/java/ru/javaops/masterjava/webapp/UsersServlet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web/webapp/src/main/java/ru/javaops/masterjava/webapp/UsersServlet.java	(date 1491066006000)
+++ web/webapp/src/main/java/ru/javaops/masterjava/webapp/UsersServlet.java	(revision )
@@ -1,9 +1,7 @@
 package ru.javaops.masterjava.webapp;
 
 import com.google.common.collect.ImmutableMap;
-import org.thymeleaf.TemplateEngine;
 import org.thymeleaf.context.WebContext;
-import ru.javaops.masterjava.common.web.ThymeleafUtil;
 import ru.javaops.masterjava.persist.DBIProvider;
 import ru.javaops.masterjava.persist.dao.UserDao;
 
@@ -14,6 +12,8 @@
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 
+import static ru.javaops.masterjava.common.web.ThymeleafListener.engine;
+
 @WebServlet("")
 public class UsersServlet extends HttpServlet {
     private UserDao userDao = DBIProvider.getDao(UserDao.class);
@@ -22,7 +22,6 @@
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
         final WebContext webContext = new WebContext(req, resp, req.getServletContext(), req.getLocale(),
                 ImmutableMap.of("users", userDao.getWithLimit(20)));
-        final TemplateEngine engine = ThymeleafUtil.getTemplateEngine(getServletContext());
         engine.process("users", webContext, resp.getWriter());
     }
 }
Index: web/export/src/main/java/ru/javaops/masterjava/export/ThymeleafListener.java
===================================================================
--- web/export/src/main/java/ru/javaops/masterjava/export/ThymeleafListener.java	(date 1491066006000)
+++ web/common-web/src/main/java/ru/javaops/masterjava/common/web/ThymeleafListener.java	(revision )
@@ -1,7 +1,6 @@
-package ru.javaops.masterjava.export;
+package ru.javaops.masterjava.common.web;
 
 import org.thymeleaf.TemplateEngine;
-import ru.javaops.masterjava.common.web.ThymeleafUtil;
 
 import javax.servlet.ServletContextEvent;
 import javax.servlet.ServletContextListener;
Index: web/export/src/main/java/ru/javaops/masterjava/export/UploadServlet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web/export/src/main/java/ru/javaops/masterjava/export/UploadServlet.java	(date 1491066006000)
+++ web/export/src/main/java/ru/javaops/masterjava/export/UploadServlet.java	(revision )
@@ -15,7 +15,7 @@
 import java.io.InputStream;
 import java.util.List;
 
-import static ru.javaops.masterjava.export.ThymeleafListener.engine;
+import static ru.javaops.masterjava.common.web.ThymeleafListener.engine;
 
 @WebServlet("/")
 @MultipartConfig
